tools/scripts/mbed.mk

```diff
@@ -44,15 +43,15 @@ SPACE := $(NULL) $(NULL)
 
 MBED_C_FLAGS = $(subst ",,$(subst $(COMMA), ,$(patsubst {"flags":[%,%,$(firstword $(subst ]$(COMMA)"macros", "macros",$(subst $(SPACE),,$(READ_MBED_GENERATED_PROFILE_C_FILE)))))))
 MBED_C_SYMBOLS = $(patsubst %,-D%,$(subst ",,$(subst $(COMMA), ,$(patsubst %]},%,$(lastword $(subst "symbols":[,"symbols":[ ,$(subst $(SPACE),,$(READ_MBED_GENERATED_PROFILE_C_FILE))))))))
-CFLAGS += $(MBED_C_FLAGS) $(MBED_C_SYMBOLS) -include $(MBED_OS_BUILD_DIRECTORY)/mbed_config.h -DMBED_PLATFORM $(TEST_FLAGS)
+CFLAGS += $(MBED_C_FLAGS) $(MBED_C_SYMBOLS) -include $(MBED_OS_BUILD_DIRECTORY)/mbed_config.h -DMBED_PLATFORM
 
 MBED_CPP_FLAGS = $(subst ",,$(subst $(COMMA), ,$(patsubst {"flags":[%,%,$(firstword $(subst ]$(COMMA)"macros", "macros",$(subst $(SPACE),,$(READ_MBED_GENERATED_PROFILE_CPP_FILE)))))))
 MBED_CPP_SYMBOLS = $(patsubst %,-D%,$(subst ",,$(subst $(COMMA), ,$(patsubst %]},%,$(lastword $(subst "symbols":[,"symbols":[ ,$(subst $(SPACE),,$(READ_MBED_GENERATED_PROFILE_CPP_FILE))))))))
```

Comment:
It seems to me all these CPP specific variables are duplicates of C specific variables so I'm wondering, why have both ? We could just as well use CFLAGS and MBED_C_FLAGS only. If we don't need to handle different CPP flags now but later, we should add this functionality when we need it.

Reply:
We need CPPFLAGS for building Mbed platform projects.

Reply:
But do they ever get modified ? Are CPPFLAGS ever different from CFLAGS ?
I can see them becoming different in the future, but what I'm saying is, in your commit they are the same, so the commit contains unnecessary modifications for now.

Reply:
If you build for mbed platform it will be different definitely now. You can try building adt7420 with PLATFORM=mbed.